lib/em/protocols/httpclient.rb

```diff
@@ -180,7 +182,7 @@ def receive_data data
           case @read_state
           when :base
             # Perform any per-request initialization here and don't consume any data.
-            @data = ""
+            @data = "".dup
```

Comment:
I haven't looked closely enough to see if it's just getting overwritten here, but the same assignment in `#post_init` (on line 92) should also be updated.  `@data` is _also_ assigned in two other places in `#receive_data`, lines 223 and 261.

Likewise, the assignment to `@content`, two lines down from here.  Curiously, `@content` is _usually_ assigned via `+=` and that probably saves it from errors, at least on the tests, but there is `<<` shovel append on line 260.

Reply:
Ah... the "two other places in `#receive_data` are both the lvar `data`, not `@data`.  The lvar isn't being mutated, so those seem fine.